qcfractal/queue/radical_adapter.py

```diff
@@ -0,0 +1,122 @@
+
+"""
+Queue adapter for Radical-Pilot
+"""
+
+import math
+import time
+import logging
+import traceback
+
+import radical.pilot as rp
+
+from typing import Any, Callable, Dict, Hashable, Optional, Tuple
+
+from qcelemental.models import FailedOperation
+
+from .base_adapter import BaseAdapter
+
+
+# ------------------------------------------------------------------------------
+#
+class RadicalAdapter(BaseAdapter):
+    """An Adapter for Radical-Pilot.
+    """
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, client: Any, logger: Optional[logging.Logger] = None, **kwargs):
+        BaseAdapter.__init__(self, client, logger, **kwargs)
+
+        self._session = client['session']
+        self._pmgr    = client['pmgr']
+        self._umgr    = client['umgr']
+        self._config  = client['config']
+        self._tasks   = list()
+        self._gc      = list()
+
+        # number of managers we can concurrently run
+        self._nmgrs = math.floor(self._config['pd']['cores'] /
+                                 self._config['mgr']['cores'])
+
+        pd = rp.ComputePilotDescription(self._config['pd'])
+        self._pilot = self._pmgr.submit_pilots(pd)
+        self._umgr.add_pilots(self._pilot)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def __repr__(self):
+        return self.client['session'].uid
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _submit_task(self, task_spec: Dict[str, Any]) -> Tuple[Hashable, Any]:
+
+        # NOTE: why is this a private method, unlike the others?
+        # TODO: bulks
+
+      # # Form run tuple
+      # func = self.get_app(task_spec["spec"]["function"])
+      # task = func(*task_spec["spec"]["args"], **task_spec["spec"]["kwargs"])
+
+        cuds = list()
+        for i in range(0, 128):
+
+            cud = rp.ComputeUnitDescription()
+            cud.executable    = '/bin/sleep'
+            cud.arguments     = ['2']
```

Comment:
`qcengine.compute(model, "program")` is the `task_spec["spec"]["args"]` field so it will be:

```python
model, program = task_spec["spec"]["args"]
cud.arguments  = [model.json(), program]
```